core/storage/trie/serialization/polkadot_codec.cpp

```diff
@@ -345,6 +398,7 @@ namespace kagome::storage::trie {
       return Error::INPUT_TOO_SMALL;
     }
     auto node = std::make_shared<BranchNode>(partial_key);
+    node->value.dirty = false;
```

Comment:
I think this logic should be encapsulated strictly inside ValueAndHash class because otherwise it's too easy to forget to update dirty from outside.